From e82c3fca5b93eb0fef07ccbad851d07c15354234 Mon Sep 17 00:00:00 2001
From: Theo GOUREAU <theo.goureau-ext@st.com>
Date: Thu, 20 Feb 2025 11:23:28 +0100
Subject: [PATCH 805/954] feat(st): add SMC error codes on-going and no-perm

Add SMC error codes STM32_SMC_ON_GOING and STM32_SMC_NO_PERM.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 33441214b5d2544fd9ce17f27cd6a7a0625d0756 feat(st): add SMC error codes on-going and no-perm]
Change-Id: I580b5f790e470f84e4b39d9436bffdff34c140bd
Signed-off-by: Theo GOUREAU <theo.goureau-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/438701
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 plat/st/common/include/stm32mp_svc_setup.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plat/st/common/include/stm32mp_svc_setup.h b/plat/st/common/include/stm32mp_svc_setup.h
index 1c542dc4f2..72b58f0d4e 100644
--- a/plat/st/common/include/stm32mp_svc_setup.h
+++ b/plat/st/common/include/stm32mp_svc_setup.h
@@ -25,6 +25,8 @@
 #define STM32_SMC_NOT_SUPPORTED		0xFFFFFFFFU
 #define STM32_SMC_FAILED		0xFFFFFFFEU
 #define STM32_SMC_INVALID_PARAMS	0xFFFFFFFDU
+#define STM32_SMC_ON_GOING		0xFFFFFFFCU
+#define STM32_SMC_NO_PERM		0xFFFFFFFBU
 
 int32_t plat_svc_smc_setup(void);
 
-- 
2.53.0

